File: client/src/types/blog.ts
export interface FAQ {
  id: string;
  question: string;
  answer: string;
}

export interface MetaTags {
  title: string;
  description: string;
  keywords: string[];
}

export interface GeoTag {
  region: string;
  placename: string;
  position: string; // lat;long format
}

export interface Blog {
  id: string;
  title: string;
  slug: string;
  description: string; // Rich HTML content
  status: "draft" | "published";
  author: string;
  featureImage: string | null;
  published_at: string | null;
  createdAt: string;
  updatedAt: string;
  faqs: FAQ[];
  meta_title: string;
  meta_description: string;
  meta_keywords: string;
  customJsonLd: string | null;
}

export interface CreateBlogInput {
  title: string;
  slug: string;
  description: string;
  status: "draft" | "published";
  published_at: string | null;
  featureImage?: string | null;
  faqs?: FAQ[];
  meta_title: string;
  meta_description: string;
  meta_keywords: string;
  customJsonLd?: string | null;
}

export interface UpdateBlogInput extends Partial<CreateBlogInput> {
  id: string;
}

File: server/app/models/blog.py
from sqlalchemy import Column, Integer, String, Text, DateTime
from datetime import datetime, timezone

from app.core.database import Base


class Blog(Base):
    __tablename__ = "blogs"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(255), nullable=False)
    description = Column(Text)
    status = Column(String(50), default="draft") 
    published_at = Column(DateTime)
    feature_image = Column(String(500))
    faqs = Column(Text)  # JSON string
    slug = Column(String(255))
    meta_title = Column(String(255))
    meta_description = Column(Text)
    meta_keywords = Column(Text)
    category = Column(String(255))
    tags = Column(Text)
    custom_json_ld = Column(Text)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(
        DateTime,
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
    )
